redis.go

```diff
@@ -48,83 +48,57 @@ func (hs *hooks) AddHook(hook Hook) {
 func (hs hooks) process(
 	ctx context.Context, cmd Cmder, fn func(context.Context, Cmder) error,
 ) error {
-	ctx, err := hs.beforeProcess(ctx, cmd)
-	if err != nil {
-		cmd.SetErr(err)
-		return err
+	var hookIndex int
+	var retErr error
+
+	for ; hookIndex < len(hs.hooks); hookIndex++ {
+		ctx, retErr = hs.hooks[hookIndex].BeforeProcess(ctx, cmd)
+		if retErr != nil {
+			cmd.SetErr(retErr)
+			break
+		}
 	}
 
-	cmdErr := fn(ctx, cmd)
-
-	if err := hs.afterProcess(ctx, cmd); err != nil {
-		cmd.SetErr(err)
-		return err
+	if retErr == nil {
+		retErr = fn(ctx, cmd)
```

Comment:
if retErr != nil -> cmd.SetErr(retErr)

Reply:
cm.SetErr occurs in fn?

Reply:
Yes, cmd.SetErr is not needed here - `fn` sets the `cmd.err`